phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1199,7 +1201,20 @@ private void filterIndexCheckerMutations(Map<TableInfo, List<Mutation>> mutation
             Entry<TableInfo, List<Mutation>> pair = mapIter.next();
             TableInfo tableInfo = pair.getKey();
             if (tableInfo.getOrigTableRef().getTable().isImmutableRows() && IndexUtil.isGlobalIndexCheckerEnabled(connection, tableInfo.getHTableName())) {
-                PTable table = PhoenixRuntime.getTable(connection, tableInfo.getHTableName().getString());
+                PTable table;
+                if (MetaDataUtil.isViewIndex(tableInfo.getHTableName().getString())) {
+                    PTable origLogicalTable = tableInfo.getOrigTableRef().getTable();
+                    if (origLogicalTable.getTenantId() == null) {
+                        //Global view index
+                        table = origLogicalTable.getIndexes().get(0);
+                    } else {
+                        //tenant specific view index
+                        PTable view = PhoenixRuntime.getTableNoCache(connection, origLogicalTable.getParentName().getString());
```

Comment:
for a tenant-specific view index you'll either need a tenant connection to do the PTable lookup, or to use the new API @abhishek-chouhan created in PHOENIX-5180. (You'd have to backport PHOENIX-5180 to 4.14.3 if you go that way, as it's currently just in 4.15 and 5.1)

Reply:
Or is connection already the tenant-specific connection here?

Reply:
Yeah, looks like it's already the tenant connection. Never mind.